### ISSUE TITLE
Set l_fixed_area=T when C grid (and remap) is used

### ISSUE THREAD

**Issue by JFLemieux73** [CICE-Consortium/CICE#832]
The code should be modified so that l_fixed_area is always set to True wtih the C grid (and remap). See issues #792, #809 and #811 for details. 

An additional test should be added to the suite ( C grid, remap l_fixed_area=T).

**Comment by JFLemieux73** [comment]
add variable in module rmap and set l_fixed_area=F or T whether B or C grid in remap init

**Comment by JFLemieux73** [comment]
I have modified the code so that l_fixed_area is set in init_remap depending of the grid. 

The baseline tests assume that  l_fixed_area=.false. So at first, I wanted to verify that the end-2-end testing procedure is BFB if l_fixed_area = .false. whatever the grid. So I tested with 

if (grid_ice == 'CD' .or. grid_ice == 'C') then
         l_fixed_area = .false.
else
         l_fixed_area = .false.
endif

It is indeed BFB:

349 measured results of 349 total results
349 of 349 tests PASSED
0 of 349 tests PENDING
0 of 349 tests MISSING data
0 of 349 tests FAILED
